### PR TITLE
Handle updated/closed PRs via webhook

### PR DESCRIPTION
**Pull request by Martchus**
* Cancel a previously scheduled product when a PR is updated
  or closed.
* Introduce the CANCELLED status of a scheduled product so this
  is visible in the UI. Jobs belonging to CANCELLED scheduled
  products might later be treated differently by the cleanup as
  well.
* Related ticket: https://progress.opensuse.org/issues/127949

### CODE REVIEW

File: t/39-scheduled_products-table.t [hunk 1]
```diff
@@ -39,6 +42,7 @@ subtest 'handling assets with invalid name' => sub {
     $scheduled_product = $scheduled_products->create(\%settings);
 
     $schema->txn_begin;
+
```
Comment (okurz) [inline t/39-scheduled_products-table.t:45]: I'd prefer to revert those unrelated whitespace changes

**Review by okurz — CHANGES_REQUESTED** [review]
Could you clarify a bit more about the introduction of the CANCELLED status for scheduled products.
1. What effect does that have on non-webhook related product scheduling?
2. It sounds like a good idea but why did we not need that yet for non-webhook related product scheduling?

**Comment by Martchus** [comment]
> What effect does that have on non-webhook related product scheduling?

None at all as so far only the webhook route allows the cancellation of a scheduled produce. We *could* allow this in general. Regardless of how the cancellation happens it means that jobs of the specific scheduled product will be cancelled and become immediate subject to cleanup. (The cleanup part hasn't been implemented yet.)

> It sounds like a good idea but why did we not need that yet for non-webhook related product scheduling?

There's a mechanism to obsolete old builds where "old build" means "jobs matching the specified parameters". We should not use that here. We simply want to obsolete the set of jobs from the previous PR version, regardless how parameters might have changed. We also don't want to obsolete production jobs that happen to use the same parameters than CI jobs. We likely also want more aggressive cleanup as PR might be updated very frequently (similar to what OBS does).

**Comment by Martchus** [comment]
I've been adding comments to explain the different statuses a scheduled product can have.

**Review by okurz — APPROVED** [review]

File: lib/OpenQA/Schema/Result/ScheduledProducts.pm [hunk 2]
```diff
@@ -874,4 +914,18 @@ sub report_status_to_github ($self, $callback = undef) {
     $vcs->report_status_to_github($github_statuses_url, {state => $state}, $id, $base_url, $callback);
 }
 
+sub cancel ($self, $reason = 'PR has changed') {
```
[the remark this answers is not in the document]
Comment (Martchus) [reply lib/OpenQA/Schema/Result/ScheduledProducts.pm:917]: You've definitely spotted a not so nice workaround here. Unfortunately this default is actually used. Maybe I can try a little bit harder to make it unnecessary.

**Comment by Martchus** [comment]
The improved descriptions look good on https://github.com/os-autoinst/os-autoinst-distri-example/pull/22 and also the other improvements regarding cancellation and restarting work.

**Review by okurz — APPROVED** [review]

**Review by kalikiana — APPROVED** [review]

**Pull request merged**